docs/sources/flow: various documentation tweaks (#2220)
grafana/agent: docs/sources/flow/concepts/components.md

## Pipelines

Most arguments for a component in a config file are constant values, such
setting a `log_level` attribute to the quoted string `"debug"`:

```river
log_level = "debug"
```

_Expressions_ can be used to dynamically compute the value of an argument at

> runtime. Among other things, expressions may be used to retrieving the value of

Reviewer: ```suggestion
runtime. Among other things, expressions can be used to retrieve the value of
```

> an environment variable (`log_level = env("LOG_LEVEL")`) or referencing the

Reviewer: ```suggestion
an environment variable (`log_level = env("LOG_LEVEL")`) or to reference the
```

> export of another component (`log_level = local.file.log_level.content`).

Reviewer: ```suggestion
exported field of another component (`log_level = local.file.log_level.content`).
```

   component, and sends collected metrics to the `prometheus.remote_write`
   component.

> The following config file reprsents the above pipeline:

Reviewer: ```suggestion
The following config file represents the above pipeline:
```

```river
// Get our API key from disk.
//
// This component has an exported field called "content", holding the content
// of the file.
//
// local.file.api_key will watch the file and update its exports any time the
// file changes.
local.file "api_key" {
  filename  = "/var/data/secrets/api-key"

>   // Mark this file as sensitive to prevent its value from being shown from the

Reviewer: ```suggestion
  // Mark this file as sensitive to prevent its value from being shown in the
```
